dm_cli/import_package.py

```diff
@@ -298,7 +309,14 @@ def package_tree_from_zip(data_source_id: str, zip_package: io.BytesIO) -> Packa
                 raise Exception(
                     f"Failed to load the file '{filename}' as a JSON document"
                 )
-            uid, alias = add_file_to_package(Path(filename), root_package, json_doc)
+            if "package.json" in file_info.filename:
+                package_name = file_info.filename.rsplit("/")[-2]
+                for node in root_package.content:
+                    if node.name == package_name:
+                        node.meta = json_doc["_meta_"]
+                filenames_to_remove.append(file_info.filename)
+            else:
+                uid, alias = add_file_to_package(Path(filename), root_package, json_doc)
```

Comment:
Could all this be removed, and instead have a check inside the "add_file_to_package".
Perhaps checking the entity type is better than the filename?

Reply:
updated to check type instead of filename.

if was difficult to move the logic into add_file_to_package since it is a recursive function.